onnxruntime/core/providers/rocm/math/softmax.cc

```diff
@@ -45,7 +45,8 @@ SPECIALIZED_SOFTMAX_HELPER_IMPL(float)
 // SPECIALIZED_SOFTMAX_HELPER_IMPL(double)
 SPECIALIZED_SOFTMAX_HELPER_IMPL(MLFloat16)
 
-// miopenSoftmaxForward/Backward doesn't support BFloat16.
+// cudnnSoftmaxForward/Backward doesn't support BFloat16.
```

Comment:
cudnn is for CUDA. On AMD, it is miOpen. Can you check whether miopenSoftmaxForward/Backward supports BF16?